src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -338,66 +333,120 @@ func KafkaAuthConfig(
 	return func() (*config.TLS, error) {
-		c := *certFile
-		k := *keyFile
-		t := *truststoreFile
+		defaultVal := func() (*config.TLS, error) {
+			conf, err := configuration()
+			if err != nil {
+				return nil, err
+			}
+			return conf.Rpk.AdminApi.TLS, nil
+		}
+		return buildTLS(
+			certFile,
+			keyFile,
+			truststoreFile,
+			"REDPANDA_ADMIN_TLS_CERT",
+			"REDPANDA_ADMIN_TLS_KEY",
+			"REDPANDA_ADMIN_TLS_TRUSTSTORE",
+			defaultVal,
+		)
+	}
+}
 
-		if c == "" {
-			c = os.Getenv("REDPANDA_TLS_CERT")
-		}
-		if k == "" {
-			k = os.Getenv("REDPANDA_TLS_KEY")
-		}
-		if t == "" {
-			t = os.Getenv("REDPANDA_TLS_TRUSTSTORE")
-		}
+func BuildKafkaTLSConfig(
+	certFile, keyFile, truststoreFile *string,
+	configuration func() (*config.Config, error),
+) func() (*config.TLS, error) {
+	return func() (*config.TLS, error) {
+		defaultVal := func() (*config.TLS, error) {
+			conf, err := configuration()
+			if err != nil {
+				return nil, err
+			}
+			return conf.Rpk.KafkaApi.TLS, nil
+		}
+		return buildTLS(
+			certFile,
+			keyFile,
+			truststoreFile,
+			"REDPANDA_TLS_CERT",
+			"REDPANDA_TLS_KEY",
+			"REDPANDA_TLS_TRUSTSTORE",
+			defaultVal,
+		)
+	}
+}
 
-		if t == "" && c == "" && k == "" {
-			return nil, nil
-		}
-		if t == "" && (c != "" || k != "") {
-			return nil, fmt.Errorf(
-				"--%s is required to enable TLS",
-				truststoreFileFlag,
-			)
-		}
-		if c != "" && k == "" {
-			return nil, fmt.Errorf(
-				"if --%s is passed, then --%s must be passed to enable"+
-					" TLS authentication",
-				certFileFlag,
-				keyFileFlag,
-			)
-		}
-		if k != "" && c == "" {
-			return nil, fmt.Errorf(
-				"if --%s is passed, then --%s must be passed to enable"+
-					" TLS authentication",
-				keyFileFlag,
-				certFileFlag,
-			)
-		}
-		tls := &config.TLS{
-			KeyFile:        k,
-			CertFile:       c,
-			TruststoreFile: t,
-		}
-		return tls, nil
+// Builds an instance of config.TLS.
+// If certFile, keyFile or truststoreFile are nil, then their corresponding
+// env vars are checked (given by certEnvVar, keyEnvVar & truststoreEnvVar).
+// If after that no value is found for any of them, the result of calling
+// defaultVal is returned.
+func buildTLS(
+	certFile, keyFile, truststoreFile *string,
+	certEnvVar, keyEnvVar, truststoreEnvVar string,
+	defaultVal func() (*config.TLS, error),
+) (*config.TLS, error) {
+	// Give priority to building the TLS config with args that were passed
+	// directly or as env vars.
+	c := *certFile
+	k := *keyFile
+	t := *truststoreFile
+
+	if c == "" {
+		c = os.Getenv(certEnvVar)
+	}
+	if k == "" {
+		k = os.Getenv(keyEnvVar)
+	}
+	if t == "" {
+		t = os.Getenv(truststoreEnvVar)
+	}
+	if t == "" && c == "" && k == "" {
+		// If the values weren't set with flags nor env vars,
+		// return the TLS config for the Admin API from the config
+		return defaultVal()
+	}
+	if t == "" && (c != "" || k != "") {
+		return nil, fmt.Errorf(
+			"--%s is required to enable TLS",
```

Comment:
Why are CA certs necessary?

Reply:
That's a bug! #1693

Reply:
They shouldn't be. Fixing that in additional commits.

Reply:
Fixed.